packages/core/src/external/cda/remove-b64.ts

```diff
@@ -37,18 +47,45 @@ export function removeBase64PdfEntries(payloadRaw: string): {
         if (comp.section.entry) {
           //eslint-disable-next-line @typescript-eslint/no-explicit-any
           comp.section.entry = toArray(comp.section.entry).filter((entry: any) => {
-            const mediaType = entry.act?.text?.["_mediaType"]?.trim().toLowerCase();
-            if (
-              (BINARY_MIME_TYPES.includes(mediaType) || mediaType == undefined) &&
-              (entry.act?.text?.["_representation"]?.trim().toLowerCase() ===
-                b64Representation.toLowerCase() ||
-                entry.organizer?.component?.observationMedia?.value?.["_representation"]
-                  ?.trim()
-                  .toLowerCase() === b64Representation.toLowerCase())
-            ) {
-              isRemovedEntries = true;
-              b64Attachments.push(entry);
-              return false;
+            if (isConcernActEntry(entry)) {
+              const act = entry.act;
+              if (
+                isBinaryMimeTypeOrUndefined(act.text?._mediaType) &&
+                isB64Representation(act.text?._representation)
+              ) {
+                isRemovedEntries = true;
+                b64Attachments.acts.push(act);
+                b64Attachments.total++;
+                return false;
+              }
+            } else if (isObservationOrganizer(entry)) {
+              const mediaComponents = getMediaObservations(entry.organizer);
+              // TODO: 2474: Apparently, some XML have B64 attachments in regular observations, so need to account for that as well
+              if (!mediaComponents?.length) return true;
+
+              const allComponents = toArray(entry.organizer.component);
+              const nonMediaComponents = allComponents.filter(
+                comp => !("observationMedia" in comp)
+              );
+              const filteredMediaComponents = mediaComponents.filter(comp => {
+                const val = comp.observationMedia.value;
+                const shouldRemove =
+                  isBinaryMimeTypeOrUndefined(val?._mediaType) &&
+                  isB64Representation(val?._representation);
+
+                if (shouldRemove) {
```

Comment:
this variable isn't necessary tbh -- just check and return false, otherwise return true...